File: binary-search/binarySearch.js
function binarySearch(arr, item) {
  let min = 0;
  let max = arr.length - 1;
  let msm = "Number not founded";

  while (min <= max) {
    let medium = Math.floor(min + max)/ 2;
    let guess = arr[medium];

    if (guess === item){
      return medium;
    }

    else if (guess > item){
      max -=1;
    }
     else {
      min +=1;
    }   
  
  }

  return msm;
}

myArr = [1,7,4,3,5,8,9];

 console.log(binarySearch(myArr, 5));
 console.log(binarySearch(myArr, 25));

File: binary-search/binary_search.rb
def binary_search(arr, item) 
  min = 0
  max = (arr.length) - 1
  msm = "Number not founded"

 while min <= max do
      mid = (min + max) / 2
      guess = arr[mid]

      if guess < item 
          min += 1
      elsif guess == item
        return mid

      elsif guess > item
        max -= 1

    
    end
  
  end
  msm
end

my_arr = [4,6,5,7,8,9,3];

print(binary_search(my_arr, 4));
